examples/refs/src/Refs.purs

```diff
@@ -18,68 +19,78 @@ import Web.HTML.HTMLElement (getBoundingClientRect)
 import Web.HTML.HTMLElement as HTMLElement
 import Web.HTML.Window as Window
 
-mkRefs :: CreateComponent {}
+mkRefs :: Effect (ReactComponent {})
 mkRefs = do
   component "Refs" \props -> React.do
-
     mouseDistance1 /\ buttonRef1 <- useNodeDistanceFromMouse
     mouseDistance2 /\ buttonRef2 <- useNodeDistanceFromMouse
     mouseDistance3 /\ buttonRef3 <- useNodeDistanceFromMouse
+    pure
+      $ fragment
+          [ element (R.unsafeCreateDOMComponent "button")
+              { ref: buttonRef1
+              , children: [ R.text $ show mouseDistance1 <> "px" ]
+              , style: R.css { width: "100px", position: "absolute", top: "20px", left: "200px" }
+              }
+          , element (R.unsafeCreateDOMComponent "button")
+              { ref: buttonRef2
+              , children: [ R.text $ show mouseDistance2 <> "px" ]
+              , style: R.css { width: "100px", position: "absolute", top: "60px", left: "40px" }
+              }
+          , element (R.unsafeCreateDOMComponent "button")
+              { ref: buttonRef3
+              , children: [ R.text $ show mouseDistance3 <> "px" ]
+              , style: R.css { width: "100px", position: "absolute", top: "120px", left: "90px" }
+              }
+          ]
 
-    pure $ fragment
-      [ element (R.unsafeCreateDOMComponent "button")
-          { ref: buttonRef1
-          , children: [ R.text $ show mouseDistance1 <> "px" ]
-          , style: R.css { width: "100px", position: "absolute", top: "20px", left: "200px" }
-          }
-      , element (R.unsafeCreateDOMComponent "button")
-          { ref: buttonRef2
-          , children: [ R.text $ show mouseDistance2 <> "px" ]
-          , style: R.css { width: "100px", position: "absolute", top: "60px", left: "40px" }
-          }
-      , element (R.unsafeCreateDOMComponent "button")
-          { ref: buttonRef3
-          , children: [ R.text $ show mouseDistance3 <> "px" ]
-          , style: R.css { width: "100px", position: "absolute", top: "120px", left: "90px" }
-          }
-      ]
+newtype UseNodeDistance hooks
+  = UseNodeDistance (UseEffect Unit (UseState Int (UseRef (Nullable Node) hooks)))
 
-type UseNodeDistance hooks = UseEffect Unit (UseState Int (UseRef (Nullable Node) hooks))
+derive instance ntUseNodeDistance :: Newtype (UseNodeDistance hooks) _
 
 useNodeDistanceFromMouse :: Hook UseNodeDistance (Int /\ (Ref (Nullable Node)))
-useNodeDistanceFromMouse = React.do
-  elementRef <- useRef null
-  mouseDistance /\ setMouseDistance <- useState 0
+useNodeDistanceFromMouse =
+  newtypeHook React.do
```

Comment:
Sorry for the reformat -- this is the main change here, plus the newtype instance above